framework/include/materials/MaterialBase.h

```diff
@@ -341,7 +346,12 @@ template <typename T>
 ADMaterialProperty<T> &
 MaterialBase::declareADProperty(const std::string & prop_name)
 {
+  // Check if the supplied parameter is a valid imput parameter key
+  std::string mat_name = prop_name;
+  if (_pars.have_parameter<MaterialPropertyName>(prop_name))
+    mat_name = _pars.get<MaterialPropertyName>(prop_name);
+
   _fe_problem.usingADMatProps(true);
-  registerPropName(prop_name, false, MaterialBase::CURRENT);
-  return materialData().declareADProperty<T>(prop_name);
+  registerPropName(mat_name, false, MaterialBase::CURRENT);
+  return materialData().declareADProperty<T>(mat_name);
```

Comment:
prop_name -> name

mat_name -> prop_name

please

Reply:
The input parameter is either a property name of an input parameter name, let's cal it `name`. A material name is not the same as a material property name, `prop_name` is the right thing here.

Reply:
so do you want `declareADProperty(const std::string & name)`?

Reply:
or just mat_name -> name?

Reply:
I did the latter since it changes less things, let me know if it makes sense to you